src/subhalo_tracking.cpp

```diff
@@ -526,16 +526,33 @@ void SubhaloSnapshot_t::FeedCentrals(HaloSnapshot_t &halo_snap)
     {
       auto &central = Subhalos[Members[0]];
       assert(central.Particles.size());
+      HBTInt tracerIndex = central.GetTracerIndex(); // Save tracerIndex before swap
       central.Particles.swap(Host.Particles); // reuse the halo particles
       central.Nbound = central.Particles.size();
+      bool tracerIndexSet = false;
+      // Use the most bound tracer from the previous snapshot that remains
+      // in the FOF group as the tracer. We need this information for the masking.
+      for (HBTInt i = tracerIndex; i < Host.Particles.size(); i++)
       {
-        auto mostbndid = Host.Particles[0].Id;
+        if (tracerIndexSet)
+          break;
+        while (!Host.Particles[i].IsTracer())
```

Comment:
Might be better to have this as an `if(!Host.Particles[i].IsTracer()) continue;`? I would avoid having a while inside a for (I think the alternative is more understandable)

Reply:
Changed